### PR TITLE
feat: populate block fields

### LINKED ISSUE #358: Correctly fill block details
When we create blocks for in-memory-node, we set [most fields to default](https://github.com/matter-labs/era-test-node/blob/main/src/node/in_memory.rs#L114).
SDKs may rely on these fields, e.g. `viem` may use `baseFeePerGas` to set `maxFeePerGas` for transactions.

We need to make sure that for each field we set a sane value that makes practical sense if anyone uses it.

### PR DESCRIPTION
**Pull request by itegulov**
# What :computer: 

Makes sure that all block fields are set to reasonable values

# Why :hand:

See #358 for motivation

### CODE REVIEW

File: src/node/in_memory.rs [hunk 1]
```diff
@@ -1461,24 +1465,40 @@ impl<S: ForkSource + std::fmt::Debug + Clone> InMemoryNode<S> {
...
             l1_batch_number: Some(U64::from(batch_env.number.0)),
+            base_fee_per_gas: U256::from(get_batch_base_fee(&batch_env, VmVersion::latest())),
+            timestamp: U256::from(batch_env.timestamp),
+            l1_batch_timestamp: Some(U256::from(batch_env.timestamp)),
             transactions: vec![TransactionVariant::Full(transaction)],
             gas_used: U256::from(tx_result.statistics.gas_used),
-            gas_limit: U256::from(BATCH_GAS_LIMIT),
+            gas_limit: U256::from(get_max_batch_gas_limit(VmVersion::latest())),
+            logs_bloom,
             ..Default::default()
```
Comment (popzxc) [inline src/node/in_memory.rs:1502]: Can we remove that and instead set the remaining fields with comments explaining why it's OK for them to be so?

**Review by popzxc — CHANGES_REQUESTED** [review]
There is also [create_empty_block](https://github.com/matter-labs/era-test-node/blob/main/src/node/in_memory.rs#L93C8-L93C26) that is used in different places, such as:
- `mine_empty_blocks`
- create fictive miniblock
- create the fist initial block.

I believe the issue with viem that led to #358 creation is caused by the last one (because I was running against fresh instance).
Probably `create_empty_block` should be either significantly reworked or removed at all.

**Comment by itegulov** [comment]
@popzxc sorry, I should have been more clear about the scope of the PR. There are indeed many problems with how we create empty blocks and I was planning to address that in a separate PR that reworks when they are added (right now every second block is empty). I will cherry pick the relevant changes here.

**Comment by popzxc** [comment]
@itegulov it's up to you -- we can have multiple granular PRs (that probably will work better, actually), no need to bring everything to a single PR. I just wanted to highlight that this PR doesn't close #358 as it doesn't solve the problem that caused the issue to be created.

**Comment by itegulov** [comment]
> I just wanted to highlight that this PR doesn't close https://github.com/matter-labs/era-test-node/issues/358 as it doesn't solve the problem that caused the issue to be created.

Ah, fair enough, I remove the wording that would close it.

**Review by popzxc — APPROVED** [review]

**Pull request merged**